backend/api/academics/section.py

```diff
@@ -98,7 +98,15 @@ def new_section(
     Returns:
         SectionDetails: Section created
     """
-    return section_service.create(subject, section)
+    created_section = section_service.create(subject, section)
+    section_service.add_lecture_room_to_section(
+        subject, created_section, section.lecture_room
+    )
+
```

Comment:
Without looking at the implementation of these service methods, I'm a little surprised and slightly confused by how this API implementation reads.

I wonder if a little refactoring or commenting could make it more obvious why the mutating call is needed after create and why the conditional statement is needed?

Reply:
@KrisJordan yeah, this was a very interesting issue that I was discovering as well. This stems from the fact that our `SectionModel` splits up `rooms` into two properties -- `lecture_room` and `office_hour_rooms`. So, when we try and POST a section model, we cannot post directly into the table for the section entity.

My solution was this super ugly implementation.. ☹️ - it relies on section being created *first* (so that its ID field is populated), then connecting it to a room in the `academics__section_room` table.

Do you think this is alright here? I will add a comments if so.